src/jit/emitxarch.cpp

```diff
@@ -4797,6 +4895,18 @@ void emitter::emitIns_AX_R(instruction ins, emitAttr attr, regNumber ireg, regNu
 }
 
 #if FEATURE_HW_INTRINSICS
+void emitter::emitIns_SIMD_R_R(instruction ins, regNumber reg1, regNumber reg2, var_types simdtype)
+{
+    emitIns_R_R(ins, emitTypeSize(simdtype), reg1, reg2);
```

Comment:
Which instructions where hitting this case?

Reply:
This is related to issue which I planned to open. My preference is to use simpler code for HW Intrinsics which are known to have 2 operands and retVal is equal to first operand. Currently it is called via `emitIns_SIMD_R_R_R` and first checks if `VEX encoding` is supported. My assumption is that this path will never be hitted for all SSE/SSSE code, therefore it is better to use simpler function.

However, there is a catch, if during runtime processor has support for AVX or AVX2 this path will be hit if developer will not implement any code checking `IsSupported` above SSE2 than some of the intrinsics which are promoted to 128bit VEX encoded AVX/AVX2 instructions can be encoded with VEX prefix and will run as an AVX/AVX2 128 bit code.

The benefit of such situation is that there is no requirement for data alignment during loading and storing with the risk that code path contains SSE only instructions what will result in high transition penalty between AVX - SSE - AVX. Currently this path raises such risks.

Functions are not hooked into code - my intention was to show implementation and discuss best solution on the example. Full implementation would still require moving `reg == reg1` and conditional code into the body.

Reply:
> My assumption is that this path will never be hitted for all SSE/SSSE code, therefore it is better to use simpler function.

Why would it not be hit for all SSE codes? Currently, the JIT will always emit the VEX encoded form of the instruction, when the underlying hardware supports it. This helps ensure that there is no transition penalty, unless calling until some P/Invoke code that is not also AVX aware. It also allows the generated code to be more optimal for the actual underlying hardware.

Reply:
@tannergooding has explained very good. Please do not worry about AVX-SSE transition penalties, we have addressed it https://github.com/dotnet/coreclr/pull/8588 https://github.com/dotnet/coreclr/pull/15014